Automattic/jetpack
# Add TypeScript support

- This PR adds TypeScript support for `js-packages`, `plugins/jetpack` and this can be extended to any part of the codebase.
- You can now write all your components in TS.
- Adds TS support for Unit tests - giving you more confidence writing tests.
- You can now even use TypeScript for Storybook stories.

Also,
- It updates the configuration files related to Unit tests, Storybook etc.
- It adds some useful types (like `@types/react` etc.) as dependencies

#### Jetpack product discussion
p9dueE-4E8-p2

#### Does this pull request change what data or activity we track or use?
No

### Testing instructions

- Convert any component inside plugins/jetpack to TypeScript
- Run `jetpack build plugin/jetpack`
- Do the same for `js-packages/components`

See #23889 as an example/POC to use TypeScript.
